challenge.go

```diff
@@ -63,7 +64,7 @@ type ChallengesSystem interface {
 	Leave(ctx context.Context, logger runtime.Logger, nk runtime.NakamaModule, userId, challengeId string) (*Challenge, error)
 
 	// SubmitScore submits a new score to the challenge.
-	SubmitScore(ctx context.Context, logger runtime.Logger, nk runtime.NakamaModule, userId, challengeId string, score, subscore int64, metadata map[string]any) (challenge *Challenge, err error)
+	SubmitScore(ctx context.Context, logger runtime.Logger, nk runtime.NakamaModule, userId, challengeId string, score, subscore int64, metadata map[string]any, conditionalMetadataUpdate bool) (challenge *Challenge, err error)
```

Comment:
nit: is the parameter named the same as the metadata related override we support with event leaderboard score submissions?

Reply:
It's the same we declared in the interface: https://github.com/heroiclabs/hiro/blob/main/event_leaderboards.go#L79
However I see that there's a discrepancy here: https://github.com/heroiclabs/hiro-gdk/blob/main/server/event_leaderboards.go#L115

Reply:
@zyro what do you suggest?

Reply:
`conditionalMetadataUpdate` is the correct name, I think this is just a place where we failed to update it https://github.com/heroiclabs/hiro-gdk/blob/main/server/event_leaderboards.go#L115 and we should definitely do so.

Reply:
I'll update as part of this PR